OpenBikeSensorFirmware/configServer.cpp

```diff
@@ -1065,6 +1069,90 @@ void startServer() {
 
   server.on("/privacy_action", privacyAction);
 
+  // ###############################################################
+  // SD card file systen access
+  // ###############################################################
+
+  server.on("/sd", []() {
+    String path = "/";
+    if (server.hasArg("path")) {
+      path = server.arg("path");
+    }
+
+    File file = SDFileSystem.open(path);
+
+    if (!file) {
+      server.send(404, "text/plain", "File not found.");
+      return;
+    }
+
+
+    if (file.isDirectory()) {
+      String html = header;
+
+      // Header
+      html.replace("{version}", OBSVersion);
+      html.replace("{subtitle}", "SD Card Contents");
+
+      html += "<ul class=\"directory-listing\">";
+
+      // Iterate over directories
+      File child = file.openNextFile();
+      while(child) {
+        html += ("<li class=\""
+          + String(child.isDirectory() ? "directory" : "file")
+          + "\"><a href=\"/sd?path="
+          + String(child.name())
+          + "\">"
+          + String(child.name()).substring(1)
+          + String(child.isDirectory() ? "/" : "")
+          + "</a></li>");
+
+        child.close();
+        child = file.openNextFile();
+      }
+
+      file.close();
+
+      html += "</ul>";
+      html += footer;
+      server.send(200, "text/html", html);
+
+      file.close();
+      return;
+    }
+
+    String dataType;
+    if (path.endsWith(".htm")) {
+      dataType = "text/html";
+    } else if (path.endsWith(".css")) {
+      dataType = "text/css";
+    } else if (path.endsWith(".js")) {
+      dataType = "application/javascript";
+    } else if (path.endsWith(".png")) {
+      dataType = "image/png";
+    } else if (path.endsWith(".gif")) {
+      dataType = "image/gif";
+    } else if (path.endsWith(".jpg")) {
+      dataType = "image/jpeg";
+    } else if (path.endsWith(".ico")) {
+      dataType = "image/x-icon";
+    } else if (path.endsWith(".xml")) {
+      dataType = "text/xml";
+    } else if (path.endsWith(".pdf")) {
+      dataType = "application/pdf";
+    } else if (path.endsWith(".zip")) {
+      dataType = "application/zip";
+    } else {
+      // arbitrary data
+      dataType = "application/octet-stream";
+    }
+
+    server.sendHeader("Content-Disposition", String("attachment; filename=\"") + file.name() + String("\""));
```

Comment:
Could set content-length?

Reply:
I think that's done by the `streamFile` below. At least, it works fine for me ;)

Reply:
![image](https://user-images.githubusercontent.com/235784/100541743-2c769300-3246-11eb-8138-83e1c2e14855.png)

Yes, it does :)